tuning/test_libtuner.py

```diff
@@ -81,320 +73,193 @@ def test_collision_handler():
+    result3.candidate_id = 0
+    result3.device_id = "device1"
 
-    assert parsed_results == expect_parsed_results
-    assert dump_list == expect_dump_list
-    assert candidate_trackers != candidate_trackers_before
-    assert candidate_trackers == expect_candidate_trackers
+    result4 = MagicMock(spec=libtuner.TaskResult)
+    result4.result = MagicMock(stdout="4.13")
+    result4.candidate_id = 0
+    result4.device_id = "device2"
 
+    candidate_results = [result1, result2]
+    baseline_results = [result3, result4]
 
-def test_parse_grouped_benchmark_results():
-    def generate_res(stdout: str, device_id: int) -> libtuner.TaskResult:
-        result = libtuner.subprocess.CompletedProcess(
-            args=[""],
-            stdout=stdout,
-            returncode=0,
-        )
-        return libtuner.TaskResult(result=result, device_id=device_id)
-
-    def set_tracker(
-        tracker: libtuner.CandidateTracker,
-        model_benchmark_time: float,
-        model_benchmark_device_id: int,
-        baseline_benchmark_time: float,
-        calibrated_benchmark_diff=float,
-    ):
-        tracker.model_benchmark_time = model_benchmark_time
-        tracker.model_benchmark_device_id = model_benchmark_device_id
-        tracker.baseline_benchmark_time = baseline_benchmark_time
-        tracker.calibrated_benchmark_diff = calibrated_benchmark_diff
-
-    b1 = "Benchmarking: some_dir/baseline.vmfb on device 0 BM_main/process_time/real_time_median 60.7 ms 13.5 ms 5 items_per_second=16.4733/s"
-    b2 = "Benchmarking: baseline.vmfb on device 1 BM_main/process_time/real_time_median 59.8 ms 15.1 ms 5 items_per_second=16.7114/s"
-    s1 = "Benchmarking: unet_candidate_1.vmfb on device 0 BM_main/process_time/real_time_median 62.4 ms 15.4 ms 5 items_per_second=16.0223/s"
-    s2 = "Benchmarking: some_dir/unet_candidate_2.vmfb on device 1 BM_main/process_time/real_time_median 61.4 ms 11.0 ms 5 items_per_second=16.2958/s"
-    s3 = "Benchmarking: unet_candidate_4.vmfb on device 1 BM_main/process_time/real_time_median 57.4 ms 11.0 ms 5 items_per_second=16.2958/s"
-
-    grouped_benchmark_results = [
-        [generate_res(b1, 0), generate_res(s1, 0)],
-        [
-            generate_res(b2, 1),
-            generate_res(None, 1),
-            generate_res(s2, 1),
-            generate_res(s3, 1),
-        ],
-    ]
-
-    path_config = libtuner.PathConfig()
+    # Mock IREEBenchmarkResult to return float value from stdout
+    def mock_get_mean_time(self):
+        return float(self.result_str)
 
-    candidate_trackers = [libtuner.CandidateTracker(i) for i in range(5)]
+    # Mock IREEBenchmarkResult to return specific benchmark times
+    with patch("libtuner.IREEBenchmarkResult.get_mean_time", new=mock_get_mean_time):
+        # Mock generate_display_MBR to return a fixed display string
+        with patch(
+            "libtuner.generate_display_MBR",
+            side_effect=lambda *args, **kwargs: "display_str",
+        ):
```

Comment:
Why do we ever have to rely on the generate_display_mbr in tests? Printing to the screen should be an implementation detail